### DISCUSSION TITLE
Gradients larger than model parameter using examples

### DISCUSSION THREAD

**Discussion by tjk9501** [slimgroup/JUDI.jl#246] [Q&A]
Hello,

I am using the RTM example to derive model update gradients
```
#x_iter is the initial slowness
r = Ml[i]*J[i]*Mr*x_iter - Ml[i]*d_obs[i]
g = adjoint(Mr)*adjoint(J[i])*adjoint(Ml[i])*r
t = norm(r)^2/norm(g)^2
global x_iter -= t*g
```
This is an wiki example, I tried with two different examples. One is using the marmousi_model and marmousi signal
```
ftp://slim.gatech.edu/data/SoftwareRelease/Imaging.jl/2DLSRTM/marmousi_model.h5 
ftp://slim.gatech.edu/data/SoftwareRelease/Imaging.jl/2DLSRTM/marmousi_2D.segy
```
The other is using my own signals. It turns out that using marmousi model (with ricket_wavelet) will yield normal gradients
```
t*g
```
smaller than x_iter, but use my own model (in fact I provide my own signal) will yield gradients
```
t*g
```
larger than iterative slowness so than update slowness will become negative. Are there any solutions to this problem? Using same signal with FWI examples in Wiki can run without any issue.

**Comment by mloubout** [comment]
You would have to use box constraints such as 

https://github.com/slimgroup/JUDI.jl/blob/master/examples/scripts/fwi_example_minConf.jl

to enforce bounds on the update.

**Reply by tjk9501** [reply]
Yes, I have used constraints like
```
proj(x) = reshape(median([vec(mmin) vec(x) vec(mmax)]; dims=2),model0.n)
```
But the updated gradients t*g is very large, even with constraints like proj(x), the iterative velocity map will mostly become maximum velocity defined in the constraint function.
One thing that I am not sure about is the amplitude of observed data, because I see the marmousi signal have maximum amplitude like 1e-3, but my own signal have amplitude that is like 1e1. I wonder if this is the reason that cause very large gradients

**Reply by mloubout** [reply]
That's what the optimization algorithm and line search are for.

**Reply by tjk9501** [reply]
A further question. I have seen the example in preconditioners Wiki. It is inverting a marmousi model. And the initial  model is defined as
```
x_iter=zeros(...)
```
I am confused about this. I think marmousi model is a complicated model. And initialize x_iter with zero slowness means infinite velocity. In FWI we usually use a close starting model, for example we want to invert 1.8km/s we start from 1.5km/s. But RTM example we strat from zero slowness. If we want to start from a close slowness, how to initialize that? I have tried to initialize x_iter with some initial slowness guess but it yields very large gradients.

**Reply by mloubout** [reply]
Is that wiki doing FWI or LSRTM?  Because one invert for velocity/slowness (FWI) while the other invert for a model perturbation  `\delta vp` (LSRTM) sand in the second case yes you would start at zero

**Reply by tjk9501** [reply]
It is a RTM. Ok, thanks.
